File: Code/Avernakis Nodejs/Avernakis-Nodejs/Private/UiProgressBar.cpp
#include "StdAfx.h"
#include "UiProgressBar.h"

#define ThisMethod($x) &UiProgressBar::$x
#define AutoAddMethod($x, ...) AddMethod<__VA_ARGS__>( #$x, ThisMethod( $x ) )
#define MakeThisFunc($x) MakeFunc( this, &UiProgressBar::$x )

namespace Nav
{

	namespace
	{
		ObjectRegister<UiProgressBar> c_obj;
	}

	void UiProgressBar::DefineControl()
	{
		AutoAddMethod( SetState );
		AutoAddMethod( GetState, WrapObjectGeneric );
		AutoAddMethod( SetMaximum );
		AutoAddMethod( GetMaximum, WrapObjectGeneric );
		AutoAddMethod( SetValue );
		AutoAddMethod( GetValue, WrapObjectGeneric );
		AutoAddMethod( SetStep );
		AutoAddMethod( GetStep, WrapObjectGeneric );
		AutoAddMethod( Step );
		AutoAddMethod( SetAnimation );
		AutoAddMethod( GetAnimation, WrapObjectGeneric );
	}

	U1 UiProgressBar::Ctor( UiWindow * p, Napi::Value v )
	{
		if ( !__CreateControl( p, v ) )
			return false;

		return true;
	}

}

File: Code/Avernakis Nodejs/Test-Nodejs/src/Ave/Ui/Control/UiProgressBar.ts
import { IControl } from "../UiControl";
import { AveLib } from "../../AveLib";
import { WindowLike } from "./UiWindow";
import { ProgressBarState } from "../UiCommon";

export interface IProgressBar extends IControl {
	new (window: WindowLike): IProgressBar;

	SetState(n: ProgressBarState): ProgressBar;
	GetState(): ProgressBarState;

	SetMaximum(n: number): ProgressBar;
	GetMaximum(): number;

	SetValue(n: number): ProgressBar;
	GetValue(): number;

	SetStep(n: number): ProgressBar;
	GetStep(): number;
	Step(): ProgressBar;

	SetAnimation(b: boolean): ProgressBar;
	GetAnimation(): boolean;
}

export class ProgressBar extends (AveLib.UiProgressBar as IProgressBar) {}
